lib/vanagon/component.rb

```diff
@@ -231,13 +231,18 @@ def get_build_dir
     # Fetches secondary sources for the component. These are just dumped into the workdir currently.
     #
     # @param workdir [String] working directory to put the source into
-    def get_sources(workdir)
+    def get_sources(workdir) # rubocop:disable Metrics/AbcSize
+      # create an empty array if @extract_with doesn't already exist
+      @extract_with = Array(@extract_with)
```

Comment:
I didn't realize you were coercing the value of `@extract_with` to an empty array, even if it's already an array. Wouldn't that be better handled by using the accessor than the instance variable, and using the accessor to ensure that `extract_with` returns what you need it to?

```ruby
# example only; didn't do the research
def extract_with
  @extract_with ||= []
end
```